index.php

```diff
@@ -4,63 +4,107 @@
 
 function prepDirectoryPath($file) {
 
-    $path = explode('/', $file);
+    //ensure that the file is not empty
+    if($file!=='' &&!empty($file) && isset($empty))
```

Comment:
you are correct. I should have not included if($file=='') no need to do that. However, we might want to keep isset as empty cannot check undefined variable.